src/Nethermind/Nethermind.Taiko/Rpc/TaikoEngineRpcModule.cs

```diff
@@ -208,6 +210,14 @@ void CommitAndDisposeBatch(Batch batch)
                         while (i < txSource.Length && txSource[i].SenderAddress == tx.SenderAddress) i++;
                         continue;
                     }
+
+                    // For Surge, filter out any transaction with very high gas limit
+                    if (surgeConfig.MaxGasLimitRatio > 0 && tx.GasLimit > tx.SpentGas * surgeConfig.MaxGasLimitRatio)
```

Comment:
Shouldn't be opposite like tx.SpentGas > tx.GasLimit * surgeConfig.MaxGasLimitRatio